restaking/Cargo.toml

```diff
@@ -0,0 +1,11 @@
+[package]
+name = "commonware-restaking"
+version = "0.1.0"
+edition = "2021"
+license = "MIT OR Apache-2.0"
+
+[dependencies]
+alloy = { version = "0.9.2", features = ["full"] }
```

Comment:
Wondering if using `alloy` intead of `ethers` is a good choice, I have not found satisfying resources to mock Contracts / Provider.

Reply:
I've heard alloy is now the preferred library for folks. Curious if either let's you use a16z's helios (https://github.com/a16z/helios) to verify results or if they assume you trust your RPC.

Reply:
Yep this is why I did not use `ethers`, it is now deprecated 👍
I don’t see a reason it would not let you interact with Elios, but I might be missing something🤔

Reply:
That makes sense for me (and aligns with my understanding of how to best test).

Its a shame the testing has to be so heavyweight for this but I think that's the best way to ensure it works reliably (unless they change the contract upstream I suppose....unless it is immutable?).

Reply:
> Its a shame the testing has to be so heavyweight for this but I think that's the best way to ensure it works reliably (unless they change the contract upstream I suppose....unless it is immutable?).

Yep it is a little bit heavy 😀 but when your build process is ready to compile `Solidity Mocked Contract`, everything is almost done.
Technically contracts could evolve, teams developing their AVS are responsible to deploy middleware contracts, these could be different in few months.

Reply:
I have added basic tests in the new commits. I think tomorrow I will look into Symbiotic in details.
Note that operator has a specific weights for each `quorum` it is part of. I need to take another look but AVS team can apply like a "coefficient" on some quorum, Hence to have a global `Weight` this would require to get these coefficients (maybe it's out of the scope right now)

Reply:
If you want to spike on a separate version of this PR that just includes Symbiotic stuff, may make it easier to understand how the interface could be unified (if we determine that even makes sense).

We may just opt to have 2 dialects with different interfaces entirely.

Reply:
It might make sense, it does not bother you if I open a second Draft to show the Symbiotic stuff?

> We may just opt to have 2 dialects with different interfaces entirely (like restaking::eigen and restaking::symbiotic)
Yep, and even if at the beginning they are distinct, at a moment a common way to use it could appear, but it will potentially be iterative 🙂